easypy/timing.py

```diff
@@ -286,6 +286,9 @@ def sleep():
             try:
                 ret = pred(is_final_attempt=bool(expired))
             except PredicateNotSatisfied as _exc:
+                if getattr(_exc, "duration", 0):
```

Comment:
What if the user sets an attribute named `duration`? Shouldn't we add an attribute - like `_already_timed_out`?